packages/sdk/src/modules/storage/index.ts

```diff
@@ -0,0 +1,248 @@
+import { ApiPromise } from '@polkadot/api';
+import type { ISubmittableResult } from '@polkadot/types/types';
+import type { CodecHash } from '@polkadot/types/interfaces/runtime/types';
+import { CreateStorageKeysEnum, Address } from '../../types';
+import { createStorageKeys } from '../../utils';
+import { stringToU8a, u8aToHex, hexToString } from '@polkadot/util';
+import { StorageError, ItemTypeError, ItemError, StorageAddressError, StorageSeedError} from '../../utils/errors';
+import type { SDKMetadata} from '../../types';
+import { Base } from '../base';
+
+type AddItemOptions = {
+    itemType: string;
+    item: string;
+    seed?: string;
+} 
+
+type RemoveItemOptions = {
+    itemType: string;
+    seed?: string;
+}
+
+type GetItemOptions = {
+    itemType: string;
+    address?: string;
+}
+
+type UpdateItemOptions = {
+    itemType: string;
+    item: string;
+    seed?: string;
+}
+
+type AddItemResult = {
+    log: string;
+    block_hash: CodecHash;
+    unsubscribe: () => void;
+}
+
+type RemoveItemResult = {
+    log: string;
+    block_hash: CodecHash;
+    unsubscribe: () => void;
+}
+
+type GetItemResult = {
+    log: string;
+}
+
+
+type UpdateItemResult = {
+    log: string;
+    block_hash: CodecHash;
+    unsubscribe: () => void;
+}
+
+export class Storage extends Base {
+    constructor(
+      protected override readonly _api?: ApiPromise,
+      protected readonly _metadata?: SDKMetadata
+    ) {
+      super();
+    }
+  
+    /**
+     * Adds item to peaqStorage
+     * 
+     * TODO Determine what constraints we need to establish before adding to storage
+     */
+
+    public async addItem(
+     options: AddItemOptions,
+     statusCallback?: (result: ISubmittableResult) => void | Promise<void>
+  ): Promise<AddItemResult> {
+    try {
+        const api = this._getApi();
+
+        const {itemType, item, seed = ''} = options;
+        // Need to know what types itemType can to ensure proper storage (for example, only converts string to byte array to check length; conditional for other types)
+        // - is there a way to do this no matter the type?
+
+        // check if object is uint8array, if it is not then convert to see how big of storage will be needed
+
+        // checks
+        if (!itemType) throw new ItemTypeError('Item Type name is required');
+        if (!item) throw new ItemError('Item name is required');
+        if (seed !== '') this._checkSeed(seed);
+
+        // convert string to bytes to count before calling extrinsics
+        if (stringToU8a(itemType).length > 64) throw new ItemTypeError('New Item Type cannot be larger than 64 bytes');
```

Comment:
performs an error check for itemType and item size to reduce the amount of extrinsics sent to the peaq node